hpfeeds/broker/auth/database.py

```diff
@@ -0,0 +1,62 @@
+import json
+import logging
+
+from hpfeeds.broker.server import ServerException
+
+try:
+    from databases import Database
+    lib_databases = True
+except ImportError:
+    lib_databases = False
+
+
+class Authenticator(object):
+
+    def __init__(self, connection_string):
+        self.logger = logging.getLogger('hpfeeds.broker.auth.databases.Authenticator')
+        self.connection_string = connection_string.lstrip("database+")
+        self.connected = False
+
+        if not lib_databases:
+            raise ServerException("The module 'databases.Database' is not available - is the databases package installed?")
+
+    async def db_connect(self):
+
+        try:
+            self.logger.debug("Connecting to database")
+            self.database = Database(self.connection_string)
+            await self.database.connect()
+            self.connected = True
+        except Exception as err:
+            raise ServerException(f"Unable to connect to database: {err}")
+
+    async def start(self):
+        pass
+
+    async def close(self):
+        await self.database.disconnect()
+
+    async def get_authkey(self, ident):
+        self.logger.debug(f"Auth key for {ident} requested")
+
+        if not self.connected:
+            await self.db_connect()
```

Comment:
self.connected is never set to False again. What happens if e.g. you authenticate with hpfeeds, then restart the postgres or mysql database (to break the connection) and then connect again? Does `databases` transparently reconect? Or does it fail with an exception? Ideally we should ask `self.database` is its connected directly perhaps?

Reply:
Added some better handling and check `databases.is_connected`

Tested with `MySQL` turning it off will raise an exception on connection attempts. Turning the Database back on and it reconnects without issue.